jwst/flatfield/flat_field.py

```diff
@@ -187,17 +189,40 @@ def apply_flat_field(science, flat):
     # For CubeModel science data, apply flat to each integration
     if isinstance(science, datamodels.CubeModel):
         for integ in range(science.data.shape[0]):
```

Comment:
Nothing to do with this particular PR, but is there a reason why the flat needs to be applied to 3-D science data by looping over each integration individually, as opposed to just letting numpy array arithmetic do the 3d-by-2d division for us in one fell swoop?

Reply:
I.e. allowing array broadcasting to do its work?  The flat is 2D and science is 3D.

Good question.  I can try it out.

Reply:
Turns out `numpy` array broadcasting takes care of this automatically.  I'll update (and simplify) the code.  Thanks for the suggestion.